src/app/components/loaders/stx-balance-loader.tsx

```diff
@@ -16,39 +21,30 @@ interface StxBalanceLoaderProps {
   ): React.ReactNode;
 }
 export function StxBalanceLoader({ address, children }: StxBalanceLoaderProps) {
-  const { filteredBalanceQuery, isLoadingAdditionalData } = useStxCryptoAssetBalance(address);
-  const { data: balance, isLoading } = filteredBalanceQuery;
-  if (!balance) return;
-  return children(balance, isLoading, isLoadingAdditionalData);
+  const balance = useStxAddressBalance(address);
+  if (!balance.value) return;
+  return children(balance.value.stx, balance.state !== 'success', balance.state !== 'success');
 }
 
-export function StxAssetItemBalanceLoader({ address, children }: StxBalanceLoaderProps) {
-  const { initialBalanceQuery, filteredBalanceQuery, isLoadingAdditionalData } =
-    useStxCryptoAssetBalance(address);
-
-  async function refetchAll() {
-    await initialBalanceQuery.refetch();
-    return filteredBalanceQuery.refetch();
-  }
-
-  if (initialBalanceQuery.isLoading) return <CryptoAssetItemPlaceholder />;
-
-  if (isErrorTooManyRequests(filteredBalanceQuery)) {
-    return (
-      <CryptoAssetItemError
-        caption="STX"
-        icon={<StxAvatarIcon />}
-        onRefetch={() => refetchAll()}
-        title="Stacks"
-      />
-    );
-  }
-
-  if (!isFetchedWithSuccess(filteredBalanceQuery)) {
+interface StxAssetItemBalanceLoaderProps {
+  accountIndex: number;
+  children(
+    balance: AddressQuotedStxBalance,
+    isLoading: boolean,
+    isLoadingAdditionalData: boolean
+  ): React.ReactNode;
+}
+export function StxAssetItemBalanceLoader({
+  accountIndex,
+  children,
+}: StxAssetItemBalanceLoaderProps) {
+  const stxBalance = useStxAccountBalance(accountIndex);
+  const isLoading = stxBalance.state === 'loading';
+  if (isLoading) return <CryptoAssetItemPlaceholder />;
+  if (stxBalance.state === 'error') {
+    logger.error('StxBalanceLoader', stxBalance.errorMessage);
```

Comment:
Would avoid the logger here as here it's part of the react render flow, so if there's an error, it'll spam the console